core/quill.js

```diff
@@ -73,7 +78,7 @@ class Quill {
     this.root.classList.add('ql-blank');
     this.root.setAttribute('data-gramm', false);
     this.scrollingContainer = this.options.scrollingContainer || this.root;
-    this.emitter = new Emitter();
+    this.emitter = new Emitter({ selectionRoot });
```

Comment:
`selectionRoot` can be a Document or ShadowRoot. If a `Document` is passed when using a webcomponent, then it will never select any of the nodes inside ShadowDOM and all of the selection/intersection validation inside of Quill breaks.